### ISSUE TITLE
`TC004` false positive

### ISSUE THREAD

**Issue by osauldmy** [astral-sh/ruff#25657]
### Summary

Typecheck-only import is recommended to move out of type-check block because of similar runtime import.
While they're 2 different imports, one for runtime, another for types.

Example to replicate:

```python
from __future__ import annotations

from typing import TYPE_CHECKING

import psycopg2.sql

if TYPE_CHECKING:
    import psycopg2.extensions


def example(cursor: psycopg2.extensions.cursor) -> None:
    sql = psycopg2.sql.SQL("select 1")
    cursor.execute(sql)
```

```bash
$ ruff check --select TC004 /tmp/bug_example.py
/tmp/bug_example.py:8:12: TC004 Move import `psycopg2.extensions` out of type-checking block. Import is used for more than type hinting.
```

Changing `import psycopg2.extensions` to `from psycopg2.extensions import cursor` and the annotation, helps.

### Version

0.15.12

**Comment by ntBre** [comment]
Thanks for the report! This is a duplicate of https://github.com/astral-sh/ruff/issues/15723.
